test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -83,12 +85,16 @@ public static void RunTest<TAnalyzer> (MemberDeclarationSyntax m, List<Attribute
 			var builder = ImmutableDictionary.CreateBuilder<string, List<string>> ();
 
 			foreach (var file in GetTestFiles ()) {
-				var dirName = Path.GetFileName (Path.GetDirectoryName (file))!;
-				if (builder.TryGetValue (dirName, out var sources)) {
+				var dirName = Path.GetDirectoryName (file);
```

Comment:
nit: Might just be me, but I find it confusing that `GetDirectoryName` gives back the directory path, not just the name. Would you mind renaming things to make this a bit more readable? For example `dirName` -> `directory`, maybe introduce a `parentDirectory` variable?